### PR TITLE
Re-enable ConnectionCanSendAndReceiveMessages + logging

### PR DESCRIPTION
**Pull request by JamesNK**
https://github.com/aspnet/SignalR/issues/1485

Testing on CI

### CODE REVIEW

**Comment by JamesNK** [comment]
Even in a loop of 1000 I can't replicate this locally, and a loop of 100 on the server. There is a corefx sockets bug that is sometimes coming up when running the test so many times but it isn't related to signalr.

I added some more logging while looking at this and made some minor fixes that I've left in.

File: src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs [hunk 1]
```diff
@@ -348,14 +348,16 @@ private static Uri CreateConnectUrl(Uri url, string connectionId)
             }
             catch (Exception ex)
             {
-                Log.ErrorStartingTransport(_logger, _transport, ex);
+                Log.ErrorStartingTransport(_logger, transport, ex);
```
Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs]: Nice catch

File: test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs [hunk 2]
```diff
@@ -196,11 +196,11 @@ public EndToEndTests(ServerFixture<Startup> serverFixture, ITestOutputHelper out
             }
         }
 
-        [Theory(Skip = "https://github.com/aspnet/SignalR/issues/1485")]
+        [Theory]
         [MemberData(nameof(TransportTypesAndTransferFormats))]
         public async Task ConnectionCanSendAndReceiveMessages(TransportType transportType, TransferFormat requestedTransferFormat)
         {
-            using (StartLog(out var loggerFactory, testName: $"ConnectionCanSendAndReceiveMessages_{transportType.ToString()}"))
+            using (StartLog(out var loggerFactory, minLogLevel: LogLevel.Trace, testName: $"ConnectionCanSendAndReceiveMessages_{transportType.ToString()}_{requestedTransferFormat.ToString()}"))
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs]: :+1:

File: src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LongPollingTransport.cs [hunk 3]
```diff
@@ -93,7 +93,17 @@ public Task StartAsync(Uri url, IDuplexPipe application, TransferFormat transfer
 
             _application.Input.CancelPendingRead();
 
-            await Running;
+            try
+            {
+                await Running;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/LongPollingTransport.cs]: This should never throw. I would remove the try catch.

File: src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/ServerSentEventsTransport.cs [hunk 4]
```diff
@@ -200,7 +200,17 @@ public Task StartAsync(Uri url, IDuplexPipe application, TransferFormat transfer
 
             _application.Input.CancelPendingRead();
 
-            await Running;
+            try
+            {
+                await Running;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/ServerSentEventsTransport.cs]: This should never throw. I would remove the try catch.

File: src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/WebSocketsTransport.cs [hunk 5]
```diff
@@ -348,10 +348,14 @@ private static Uri ResolveWebSocketsUrl(Uri url)
             {
                 await Running;
             }
-            catch
+            catch (Exception ex)
             {
+                Log.TransportStopped(_logger, ex);
                 // exceptions have been handled in the Running task continuation by closing the channel with the exception
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/Internal/WebSocketsTransport.cs]: We should make this consistent with the rest and make it never throw (tests would need to be updated)

**Pull request merged**